File: examples/simp/Python/evaluator/evaluator1.py
# Import the library
import LinkPredPython as lpp
import sys # For printing
nbRuns = 10;
# Create an evaluator object
ev = lpp.Evaluator();
ev.addCNE();
ev.addADA();
ev.addKAB("KAB-2", 2); # Horizon limit = 2
ev.addKAB("KAB-4", 4); # Horizon limit = 4
ev.addROC();
ev.addTPR();
ev.run("Zakarays_Karate_Club.edges", nbRuns);
# Print the header row
res = ev.getPerfRes(0);
for r in res:
	sys.stdout.write(r.name + "\t");
sys.stdout.write("\n");
# Print the results of each iteration
for i in range(nbRuns):
	res = ev.getPerfRes(i);
	for r in res:
		sys.stdout.write("{:.4f}".format(r.res) + "\t");
	sys.stdout.write("\n");

File: examples/simp/CPP/evaluator/evaluator1.cpp
#include <linkpred.hpp>
#include <iostream>

using namespace LinkPred::Simp;

int main() {
	int nbRuns = 10;
	Evaluator eval;
	eval.addCNE();
	eval.addADA();
	eval.addKAB("KAB-2", 2); // Horizon limit = 2
	eval.addKAB("KAB-4", 4); // Horizon limit = 4
	eval.addROC();
	eval.addTPR();
	eval.run("Zakarays_Karate_Club.edges", nbRuns);
	// Print the header row
	auto res = eval.getPerfRes(0);
	for (auto it = res.begin(); it != res.end(); ++it) {
		std::cout << it->name << "\t";
	}
	std::cout << "\n";
	// Print the results of each iteration
	for (int i = 0; i < nbRuns; i++) {
		auto res = eval.getPerfRes(i);
		for (auto it = res.begin(); it != res.end(); ++it) {
			std::cout << it->res << "\t";
		}
		std::cout << "\n";
	}
	return 0;
}
